core/src/main/scala/Config.scala

```diff
@@ -375,24 +385,21 @@ object Config {
 
   private[this] val log = Logger[Config.type]
 
-  def readConfig(cfg: KConfig, http: Http, xa: DatabaseConfig => Transactor[IO]): IO[NelsonConfig] = {
+  def readConfig(cfg: KConfig, httpBuilder: BlazeClientConfig => IO[Client[IO]], xa: DatabaseConfig => Transactor[IO]): IO[NelsonConfig] = {
     // TIM: Don't turn this on for any deployed version; it will dump all the credentials
     // into the log, so be careful.
     // log.debug("configured with the following knobs:")
     // log.debug(cfg.toString)
 
-    val timeout = cfg.require[Duration]("nelson.timeout")
-
-    val http0: Http = http.configure(
-      _.setAllowPoolingConnection(true)
-      .setConnectionTimeoutInMs(timeout.toMillis.toInt))
+    val timeout = cfg.require[FiniteDuration]("nelson.timeout")
+    val http = httpBuilder(BlazeClientConfig.defaultConfig.copy(requestTimeout = timeout))
 
     val pools = Pools.default
 
     val nomadcfg = readNomad(cfg.subconfig("nelson.nomad"))
 
     val gitcfg = readGithub(cfg.subconfig("nelson.github"))
-    val git = new Github.GithubHttp(gitcfg, http0)
+    val git = http.map(client => new Github.GithubHttp(gitcfg, client, timeout, pools.defaultExecutor))
```

Comment:
Reminder to my future self debugging this - I believe this will create a new HTTP client each time `git` (or `slack`) is used which may explain the slowdown and number of connections I see being created in the logs. Should probably/definitely fix this.